src/DetailPage.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { FunctionComponent } from 'react';
+import { RouteComponentProps, Route } from 'react-router-dom';
+import DetailView from './DetailView';
+import Card from './Card';
+import {Pokemon} from './types/index'
+
+interface DetailPageProps {
+   allPokemon: Array<Pokemon>;
+};
+
+const DetailPage:FunctionComponent<DetailPageProps> =
```

Comment:
I would expect `<DetailPageProps & RouteComponentProps>` to work. But I see you have RouteComponentProps imported so did you already try that?

Reply:
I may have... it's been a hot second since I ran into this wall. When I implement your suggestion, I hit a compilation error: 

```
TypeScript error in /Users/lmiller4/Development/projects/pokedex-react-frontend/src/App.tsx(33,12):
Type '{ allPokemon: never[]; }' is missing the following properties from type 'RouteComponentProps<{}, StaticContext, any>': history, location, match  TS2739

    31 |         </Route>
    32 |         <Route path="/pokemon/:pokemonNum">
  > 33 |           <DetailPage allPokemon={allPokemon} />
       |            ^
    34 |         </Route>
    35 |       </Switch>
    36 |     </div>
```

I've been wrestling with this issue for over an hour so I'm going to step away for a moment, work on another React project, read and practice simpler implementations of Typescript, and come back to this. Let me know if you have any ideas!

Reply:
Please push up your latest. I cannot reproduce that error with the branch as it is now.

Reply:
Pushed!